pkg/network/buffer.go

```diff
@@ -1,19 +1,17 @@
 package network
 
-import "math"
-
-const minBufferSize = 256
-
 // ConnectionBuffer encapsulates a resizing buffer for ConnectionStat objects
 type ConnectionBuffer struct {
-	buf []ConnectionStats
-	off int
+	buf           []ConnectionStats
+	off           int
+	minBufferSize int
 }
 
 // NewConnectionBuffer creates a ConnectionBuffer with initial size `size`.
 func NewConnectionBuffer(size int) *ConnectionBuffer {
 	return &ConnectionBuffer{
-		buf: make([]ConnectionStats, int(math.Max(float64(size), minBufferSize))),
+		buf:           make([]ConnectionStats, size),
+		minBufferSize: size,
```

Comment:
This change will increase the minimum amount of memory system-probe will use, since it can no longer shrink to a size of 256. Perhaps an additional parameter is in order?

Reply:
`NewConnectionBuffer` now accepts an initial size and a minimum size